spec/jobs/notify_slack_spec.rb

```diff
@@ -39,6 +39,20 @@
 
       expect(response[0]).to eq("success")
     end
+
+    it "should notify if user have permission to see the post" do
+      topic
+      DiscourseSlack::Slack.set_filter_by_id(category.id, "#general", "follow")
+      Guardian.any_instance.expects(:can_see?).with(post).returns(false)
```

Comment:
Let's not use a stub here. I prefer to just use an actual user object to test it. Stubs are fragile and isn't testing the underlying code properly.